packages/ui-kit/src/hooks/use-selected-path.ts

```diff
@@ -0,0 +1,82 @@
+/**
+ * Custom hook that manages (for multi paths/languages components) the selected path state and provides a function to update it.
+ * The selected path is stored in local storage and can be updated by dispatching a custom event.
+ * The selected path is also reflected in the URL query string. If a query string activePath parameter is present on page load,
+ * then it overrides the local storage value.
+ * The reason for this double mechanism is that local storage event listeners are not triggered by the same window
+ * that updates the value. See https://developer.mozilla.org/en-US/docs/Web/API/Window/storage_event
+ *
+ * @returns An object containing the selectedPath state and the updateSelectedPath function.
+ */
+import { useState, useEffect } from 'react';
+
+const PATH_LOCAL_STORAGE_KEY = 'selected_path';
+const PATH_CHANGE_EVENT = 'selectedPathChange';
+
+const useSelectedPath = () => {
+  const [selectedPath, setSelectedPath] = useState<string | null>(null);
+
+  useEffect(() => {
+    // Read the initial value from local storage
+    const urlParams = new URLSearchParams(window.location.search);
+    const activePath = urlParams.get('activePath');
```

Comment:
WDYT about also extracting the const string so its available to the whole file and re-use this below LOC 65

Reply:
👍